src/util/util.c

```diff
@@ -971,6 +971,36 @@ bool is_user_or_group_name(const char *sudo_user_value)
     return true;
 }
 
+bool is_syntactically_correct_domain_name(const char *domain)
+{
+    size_t i;
+    char c;
+
+    if ((domain == NULL) || (domain[0] == 0)) {
+        return false;
+    }
+
+    /* Per man page:
+     * ```
+     * A domain name should only consist of alphanumeric ASCII characters,
+     * dashes, dots and underscores.
+     * ```
+     */
+    for (i = 0; i < strlen(domain); ++i) {
+        c = domain[i];
+        if (!( ((c >= '0') && (c <= '9')) ||
```

Comment:
I am thinking if there is more optimal way of write this using some bit operations. Still not sure if it is worth it as it may end up being less readable for developer.

Reply:
"Premature optimization is the root of all evil".
This function is going to be called once during program startup for param validation. It doesn't make sense to sacrifice readability for the sake of performance here.

Reply:
I like this statement :)

Reply:
While talking about readability, is there is reason why you didn't use `isalnum()` to replace the first three lines of the check?

Reply:
> While talking about readability, is there is reason why you didn't use `isalnum()` to replace the first three lines of the check?

No, there is no. Thanks, I'll update this.

Reply:
btw, I find `isalnum()` function name a little bit misfortunate as I read is "is-all-nums" (instead of is-alhpa-or-digit)